src/main/java/stsjorbsmod/actions/HealOnFatalAttackAction.java

```diff
@@ -0,0 +1,39 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.HealAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.powers.MinionPower;
+import com.megacrit.cardcrawl.vfx.combat.FlashAtkImgEffect;
+
+public class HealOnFatalAttackAction extends AbstractGameAction {
+    private DamageInfo info;
+    private int heal;
+
+    public HealOnFatalAttackAction(AbstractCreature target, DamageInfo info, int healAmount) {
+        this.info = info;
+        this.setValues(target, info);
+        this.heal = healAmount;
+        this.actionType = ActionType.DAMAGE;
+        this.duration = 0.1F;
+
+    }
+
+    public void update() {
+        if (this.duration == 0.1F && this.target != null) {
+            AbstractDungeon.effectList.add(new FlashAtkImgEffect(this.target.hb.cX, this.target.hb.cY, AttackEffect.NONE));
+            this.target.damage(this.info);
+            if ((this.target.isDying || this.target.currentHealth <= 0) && !this.target.halfDead && !this.target.hasPower(MinionPower.POWER_ID)) {
```

Comment:
My understanding is that we generally want CULL's on-kill effects to trigger even for minion kills, since it already has a character-level anti-stalling mechanic with Manifest